tools/src/icon4py/tools/py2fgen/wrapper_utils.py

```diff
@@ -119,3 +128,26 @@ def int_array_to_bool_array(int_array: np.typing.NDArray) -> np.typing.NDArray:
     """
     bool_array = int_array != 0
     return bool_array
+
+
+def as_field(  # type: ignore[no-untyped-def] # CData type not public?
```

Comment:
The previous implementation did more on the codegen side, avoiding a few Python indirections.
When we measure overhead we could go back to do more on the codegen.